src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonNuGetProject.cs

```diff
@@ -298,7 +298,7 @@ private JObject GetJson()
         {
             try
             {
-                using (var streamReader = new StreamReader(_jsonConfig.OpenRead()))
+                using (var streamReader = new StreamReader(_jsonConfig.Open(FileMode.Open, FileAccess.Read, JsonPackageSpecReader.share)))
```

Comment:
maybe make this a static public utility method on the JsonPackageSpecReader class?